docs/cartesian-product.adoc

```diff
@@ -300,37 +300,190 @@ To demonstrate with a table:
+"Conflicting parameters" means your test method has a parameter that should be injected by JUnit (e.g.: `TestReporter`) but you also try to inject it.
+
+Examples of badly configured tests/static factory method:
+
+[source,java]
+----
+class BadExamples {
+	@CartesianTest
+	@CartesianTest.Factory("resolveParameters")
+	void tooFewParameters(String string, int i, boolean b) {
+		// fails because the boolean parameter is not resolved
+	}
+
+	@CartesianTest
+	@CartesianTest.Factory("resolveParameters")
+	void tooManyParameters(String string) {
+		// fails because we try to supply a non-existent integer parameter
+	}
+
+	@CartesianTest
+	@CartesianTest.Factory("resolveParameters")
+	void wrongOrderParameters(int i, String string) {
+		// fails because the static factory method declared parameter sets in the wrong order
+	}
+
+	@CartesianTest
+	@CartesianTest.Factory("resolveTestReporterParam")
+	void conflictingParameters(String string, TestReporter info) {
+		// fails because both the factory method and JUnit tries to inject TestReporter
+	}
+
+	static Sets resolveParameters() {
+		return new Sets()
+			.add("A", "B", "C")
+			.add(1, 2, 3);
+	}
+
+	static Sets resolveTestReporterParam() {
+		return new Sets()
+			.add("A", "B", "C")
+			.add(new MyTestReporter()); // in this case MyTestReporter implements TestReporter
+	}
+}
+----
+
 == Writing your own `@ArgumentsSource` for `@CartesianTest`
 
-You might find that the available `@ArgumentsSource` annotations do not fit your need.
-In that case, you can write a custom https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsSource.html[`@ArgumentsSource`] and a corresponding https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsProvider.html[`ArgumentsProvider`] for `@CartesianTest`.
-Because `CartesianTest` works with parameters, you will have to use `CartesianArgumentsProvider` instead of `ArgumentsProvider`.
-This will let you initialize your argument source with it's corresponding parameter.
+You might find that the available annotations do not fit your need.
+For a `@ParameterizedTest`, you can write a custom https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsSource.html::[`@ArgumentsSource`] and a corresponding https://junit.org/junit5/docs/current/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ArgumentsProvider.html::[`ArgumentsProvider`].
+You can do the same thing for `@CartesianTest`, with the following caveats:
+
+ - Instead of `@ArgumentsSource`, you have to use https://javadoc.io/static/org.junit-pioneer/junit-pioneer/1.5.0/org/junitpioneer/jupiter/cartesian/CartesianArgumentsSource.html::[`@CartesianArgumentsSource`]
+ - Instead of `ArgumentsProvider`, you have to use one of two interfaces, each explored in their own section.
```

Comment:
There's a later comment in this review where I give an opinion on when and how to mention `ArgumentsProvider`. As to how that applies here: I think this explanation here is good and should stay as is, but all other mentions of `ArgumentsProvider` in this documentation should be removed.

Reply:
I think this is the _only_ explanation that's needed.